apps/src/code-studio/pd/foorm/FoormEditorManager.jsx

```diff
@@ -0,0 +1,103 @@
+// Main page for Foorm Editor interface. Will initially show a choice
+// between loading an existing configuration or an empty configuration.
+// After that choice is made, will render FoormEditor with the chosen configuration.
+
+import React from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import {Button, DropdownButton, MenuItem} from 'react-bootstrap';
+import FoormEditor from './FoormEditor';
+
+class FoormEditorManager extends React.Component {
+  static propTypes = {
+    updateFormQuestions: PropTypes.func,
+    populateCodeMirror: PropTypes.func,
+    resetCodeMirror: PropTypes.func,
+    formNamesAndVersions: PropTypes.array,
+
+    // populated by redux
+    formQuestions: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formKey: 0,
+      formPreviewQuestions: null,
+      formattedConfigurationOptions: this.getFormattedConfigurationDropdownOptions(
+        this.props.formNamesAndVersions
+      ),
+      showCodeMirror: false,
+      formName: null,
+      formVersion: null
+    };
+  }
+
+  getFormattedConfigurationDropdownOptions(formNamesAndVersions) {
+    return formNamesAndVersions.map((formNameAndVersion, i) => {
+      const formName = formNameAndVersion['name'];
+      const formVersion = formNameAndVersion['version'];
+      return (
+        <MenuItem
+          key={i}
+          eventKey={i}
+          onClick={() => this.loadConfiguration(formName, formVersion)}
+        >
+          {`${formName}, version ${formVersion}`}
+        </MenuItem>
+      );
+    });
+  }
+
+  loadConfiguration(formName, formVersion) {
```

Comment:
What happens on failure? Probably doesn't matter for now since just a V1 admin tool?

Reply:
on failure it would not update/show the editor...should make it a little friendlier than that

Reply:
Updated so if there's an error there is a message "Could not load the selected survey." and an empty editor